file/src/test/java/org/apache/calcite/adapter/file/FileAdapterTest.java

```diff
@@ -619,6 +621,72 @@ private String range(int first, int count) {
     return sb.append(')').toString();
   }
 
+  @Test void testDecimalType() {
+    sql("sales-csv", "select BUDGET from sales.\"DECIMAL\"")
+        .checking(resultSet -> {
+          try {
+            ResultSetMetaData metaData = resultSet.getMetaData();
+            assertEquals("DECIMAL", metaData.getColumnTypeName(1));
+            assertEquals(18, metaData.getPrecision(1));
+            assertEquals(2, metaData.getScale(1));
+          } catch (SQLException e) {
+            throw TestUtil.rethrow(e);
+          }
+        })
+        .ok();
+  }
+
+  @Test void testDecimalTypeArithmeticOperations() {
```

Comment:
Can you also test cases for negative cases and out of space numbers? For example a CSV has 0.0000000000001 or 10000 but the it is declared as DECIMAL(4,1).

Reply:
Done. This has been added in the new CsvEnumeratorTest file as part of testing the parseDecimal method.